### ISSUE TITLE
[0.9.6.1] `failed to store statistics` when DROP DATABASE

### ISSUE THREAD

**Issue by zstyblik** [influxdata/influxdb#5164]
Hello,

this is probably not good:

````
Dec 18 17:32:32 coreos01 docker[18059]: [query] 2015/12/18 17:32:32 DROP DATABASE testdb
Dec 18 17:32:44 coreos01 docker[18059]: [monitor] 2015/12/18 17:32:44 failed to store statistics: timeout
Dec 18 17:32:54 coreos01 docker[18059]: [monitor] 2015/12/18 17:32:54 failed to store statistics: timeout
Dec 18 17:33:04 coreos01 docker[18059]: [monitor] 2015/12/18 17:33:04 failed to store statistics: timeout
[...]
Dec 18 17:33:34 coreos01 docker[18059]: [http] 2015/12/18 17:33:34 10.0.3.1 - - [18/Dec/2015:17:32:32 +0000] GET /query?q=DROP+DATABASE+%22testdb%22&db=_internal HTTP/1.1 200 40 http://10.0.3.2:7083/? Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.80 Safari/537.36 51b05314-a5ad-11e5-8001-000000000000 1m1.94522685s
```

Observed with TSM1 at 0.9.6.1. I believe `testdb` was rather a large one.

Cheers!

**Comment by jwilder** [comment]
I think this might be the `tsdb.Store` holding a write lock for too long while the database shards are closed and data is actually deleted.  It took `1m` to drop that database.  

https://github.com/influxdb/influxdb/blob/master/tsdb/store.go#L151-L170

**Comment by jwilder** [comment]
Fixed via #6623
